### PR TITLE
configmapgenerator example

### PR DESCRIPTION
**Pull request by monopole**

### CODE REVIEW

File: cmd/kustomize/demos/configmaps.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+### Make the overlay for _development_
+
+Make the direcories:
+
+<!-- @overlays @test -->
+```
+OVERLAYS=$DEMO_HOME/overlays
+mkdir -p $OVERLAYS/development
+mkdir -p $OVERLAYS/production
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: Should we move this to the production overlay setup steps ?

File: cmd/kustomize/demos/configmaps.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  - secret.properties
+EOF
+```
+
+One can now generate the configmaps for productiont:
+
+<!-- @runProd @test -->
+```
+kustomize build $OVERLAYS/production
+```
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: Do we want to cover "hash" suffix aspects of configmap here ? If we do, that will complete the configmap documentation completely.

Comment (monopole) [reply cmd/kustomize/demos/configmaps.md]: Done, PTAL in the new section:  `Check the ConfigMap name`

sorry i squashed the commits

File: cmd/kustomize/demos/configmaps.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+The name of the generated `ConfigMap` is visible in this
+output.
+
+The name should be something like `dev-my-configmap-b5m75ck895`:
+
+ * `"dev-"` comes from the `namePrefix` field,
+ * `"my-configmap"` comes from the `configMapGenerator/name` field,
+ * `"-b5m75ck895"` comes from a deterministic hash that `kustomize`
+    computes from the contents of the map.
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: map --> configMap ?

File: cmd/kustomize/demos/configmaps.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+The hash suffix is critical.  If the map content
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: map --> configMap ?

File: cmd/kustomize/demos/configmaps.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+A deployment has no means to automatically know when or
+if a configMap in use by the deployment changes.
+
+If one changes a configMap without changing its name
+and all references to that name, one must imperatively
+restart the cluster to pick up the change.
+
+The best practice is to treat configMaps as immutable.
+
+Instead of editting configMaps, modify your declarative
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: s/editting/editing

File: cmd/kustomize/demos/configmaps.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+The name should be something like `dev-my-configmap-b5m75ck895`:
+
+ * `"dev-"` comes from the `namePrefix` field,
+ * `"my-configmap"` comes from the `configMapGenerator/name` field,
+ * `"-b5m75ck895"` comes from a deterministic hash that `kustomize`
+    computes from the contents of the map.
+
+The hash suffix is critical.  If the map content
+changes, so does the map name, along with all
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: map --> configMap ?

File: cmd/kustomize/demos/configmaps.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+
+The best practice is to treat configMaps as immutable.
+
+Instead of editting configMaps, modify your declarative
+specification of the cluster's desired state to
+point deployments to _new_ maps with _new_ names.
+`kustomize` makes this easy with its
+`configMapGenerator` directive and associated naming
+controls.  A GC process in the k8s master eventually
+deletes unused maps.
```
Comment (droot) [inline cmd/kustomize/demos/configmaps.md]: maps --> configMaps ?

File: cmd/kustomize/demos/configmaps.md (hunk 3 above)
Comment (monopole) [inline cmd/kustomize/demos/configmaps.md]: done

File: cmd/kustomize/demos/configmaps.md (hunk 4 above)
Comment (monopole) [reply cmd/kustomize/demos/configmaps.md]: done

File: cmd/kustomize/demos/configmaps.md (hunk 6 above)
Comment (monopole) [reply cmd/kustomize/demos/configmaps.md]: done

File: cmd/kustomize/demos/configmaps.md (hunk 5 above)
Comment (monopole) [reply cmd/kustomize/demos/configmaps.md]: done

File: cmd/kustomize/demos/configmaps.md (hunk 7 above)
Comment (monopole) [reply cmd/kustomize/demos/configmaps.md]: done

**Pull request merged**